content/post/087-deprecating-the-old-docs-site.md

```diff
@@ -0,0 +1,53 @@
+---
+date: 2020-04-01
+url: 2020-04-01-deprecating-the-old-docs-site
+title: Deprecating the old docs site
+author: Johnny Matthews
+---
+
+# Goodbye, IPFS legacy docs
+
+The IPFS team is deprecating the old documentation website. That’s just a fancy way of saying we’re replacing the current docs.ipfs.io website with the new [docs-beta.ipfs.io website](https://docs-beta.ipfs.io).
+
+## We built a new docs site
+
+Work on a new documentation site for IPFS started in Q3 of 2019. Before that, the docs site was built upon Hugo. While Hugo is an excellent static-site generator, it lacks an encapsulated component model, relies upon old web development conventions, and requires DOM manipulation if you want to have any kind of site interactivity.
+
+So with all that in mind, the IPFS docs and UX working group built a new site to host the documentation. The site build was finished towards the end of 2019, and a beta version was made available to the public. Call-outs were placed on the current IPFS docs site to encourage readers to test the new website. You can see a preview of the site at [docs-beta.ipfs.io](https://docs-beta.ipfs.io).
+
+## The deprecation plan
+
+We’ve built the site, put it online, and it’s been well received. Next comes the tricky part: removing the old website and putting this one in its place.
+
+Back in the early days of the internet, deprecating a site was as simple as putting up a notice saying things were going to be down for an hour or so, deleting everything in `/var/www/html`, and replacing it with the new website. Sadly, the good-old-days are gone, and we now live in a world full of checks, balances, and redirects. With that in mind, here is our plan to deprecate the old site and replace it with the new one.
+
+### Inform the world
+
+The first step is to let everyone know what we’re going to be doing. This post serves as part of the notification process, along with call-outs on the legacy site, and in the `ipfs/docs` GitHub repository.
+
+### Plan for redirects
+
+There have been several changes to the structure and naming conventions in the docs repo. Significant URL redirects must be put in place from legacy links to where the content now lives. The idea here is to minimize the amount of 404 errors encountered by the reader.
```

Comment:
We don't break links - that was one of the founding principles of IPFS. Can we do a full audit of all links that exist today (in either docs site) so we can ensure they're all mapped over to reasonable places in the new docs structure? (ex, heading links should redirect to the top-level article, where we've combined articles or broken them apart we should make a logical choice. We should NOT drop users on sad 404s)

Reply:
We’ve already been doing this; note that pages on the legacy site all have callouts directing users to the _specific_ equivalent content, and vice versa.

Reply:
> implementing these equivalent-content links as redirects.

Yeah that's what I was going for. I'll change the last sentence to: _This is to ensure that nobody ends up on a `404` page, and is always directed to the content that they want to see._